lib/app/dispatcher.rb

```diff
@@ -86,10 +86,13 @@ def start_polling_jobs!
         scheduler.when_triggered do |connector_settings, task|
           case task
           when :sync
-            # update connector sync_now flag
-            Core::ElasticConnectorActions.update_connector_sync_now(connector_settings.id, false)
-
+            # enqueue job before reset sync_now flag, to fill in the trigger_method
```

Comment:
I'm not sure about this - since it's a distributed system with multiple hosts potentially handling it, then multiple producers can enqueue a job this way.

It's important to first update sync_now to avoid double triggers.